lambda/format-activity-log/format-activity-log.ts

```diff
@@ -90,23 +56,33 @@ export const sendSqsMessage = async (
   return result.MessageId;
 };
 
-export const handler = async (event: SQSEvent): Promise<void> => {
+export const handler = async (event: DynamoDBStreamEvent): Promise<void> => {
   const { OUTPUT_QUEUE_URL, DLQ_URL } = process.env;
   const { Records } = event;
 
   await Promise.all(
     Records.map(async (record) => {
       try {
-        const userActivityLog: UserActivityLog = JSON.parse(record.body);
-        validateUserActivityLog(userActivityLog);
-        const formattedRecord = formatIntoActivitLogEntry(userActivityLog);
-        const messageId = await sendSqsMessage(
-          JSON.stringify(formattedRecord),
-          OUTPUT_QUEUE_URL
-        );
-        console.log(`[Message sent to QUEUE] with message id = ${messageId}`);
+        const txmaEvent = unmarshall(
+          record.dynamodb?.NewImage?.event.M as {
+            [key: string]: AttributeValue;
+          }
+        ) as TxmaEvent;
+        if (allowedTxmaEvents.includes(txmaEvent.event_name)) {
+          validateTxmaEventBody(txmaEvent);
+          const formattedRecord = formatIntoActivityLogEntry(txmaEvent);
+          const messageId = await sendSqsMessage(
+            JSON.stringify(formattedRecord),
+            OUTPUT_QUEUE_URL
+          );
+          console.log(`[Message sent to QUEUE] with message id = ${messageId}`);
+        } else {
+          console.log(
+            `DB stream sent a ${txmaEvent.event_name} event. Irrelevant for activity log so ignoring`
+          );
+        }
       } catch (err) {
-        const messageId = await sendSqsMessage(record.body, DLQ_URL);
+        const messageId = await sendSqsMessage(JSON.stringify(record), DLQ_URL);
```

Comment:
What happens if this call fails?  Do we lose the error logging below and the details of the event that we couldn't handle?

Reply:
Logging that a failure happens before attempting to send to DLQ, with error, but not including the associated event ID, as this would reqquire unmarshalling which can also fail. Also not including the entire DB stream record as potentially sensitive data.